homeworks/nikolay.krasniuk_Kingstar22/5-rxjs/script.js

```diff
@@ -0,0 +1,38 @@
+const { fromEvent } = window.rxjs;
+const {
+    map, throttleTime, pairwise, filter, distinctUntilChanged,
+} = window.rxjs.operators;
+
+const dynamicMenuEl = document.querySelector('[data-header-menu]');
+const hamburgerEl = document.querySelector('[data-header-burger]');
+const menuEl = document.querySelector('[data-menu]');
+const fixedMenu = document.querySelector('body');
+
+const scrollStream$ = fromEvent(window, 'scroll');
+
+const result = scrollStream$.pipe(
+    throttleTime(200),
+    map(() => window.pageYOffset),
+    pairwise(),
+    filter(([prevScroll, currentScroll]) => Math.abs((prevScroll - currentScroll)) > 50),
+    map(([prevScroll, currentScroll]) => prevScroll > currentScroll),
+    distinctUntilChanged(),
+);
+result.subscribe((event) => {
```

Comment:
It would be a little better if you subscribed immediately after you piped the stream. You don't need to create a result variable and then subscribe to it.

Reply:
done